jdaviz/core/helpers.py

```diff
@@ -402,6 +405,72 @@ def show_in_new_tab(self, title=None):  # pragma: no cover
                       DeprecationWarning)
         return self.show(loc="sidecar:tab-after", title=title)
 
+    def get_data(self, data_label, cls='default', subsets_to_apply=None, statistic=None):
+        """
+        Returns data with name equal to `data_label` of type `cls` with subsets applied from
+        `subsets_to_apply`.
+
+        Parameters
+        ----------
+        cls : `~specutils.Spectrum1D`, 'default'
+            The type that data will be returned as.
+        data_label : str
+            Provide a label to retrieve a specific data set from data_collection.
+        subsets_to_apply : str, list, optional
+            Subsets that are to be applied to data before it is returned. If a string
+            is given, it will be turned into a list and looped through. A future implementation
+            will allow applying multiple subsets to data before it is returned.
+        statistic : {'minimum', 'maximum', 'mean', 'median', 'sum', 'percentile'}, optional
+            The statistic to use to collapse the dataset
+
+        Returns
+        -------
+        data : dict
+            A dict of the transformed Glue subset objects, with keys
+            representing the subset name and values as astropy regions
+            objects.
+
+        """
+        if data_label not in self.app.data_collection:
+            return
+
+        # Follow-up effort will be to have multuple subsets apply,
+        # so we should keep this a list.
+        if isinstance(subsets_to_apply, str):
+            subsets_to_apply = [subsets_to_apply]
+
+        # Previously, this would use viewer.default_class. Do we want to pass
+        # a viewer reference in order to enable this? Or move this to the base
+        # viewer class?
+        cls = Spectrum1D if cls == 'default' else cls
+        data = self.app.data_collection[data_label].get_object(cls=cls, statistic=statistic)
+
+        if not subsets_to_apply:
+            return data
+
+        # Loop through each subset
+        for subsets in self.app.data_collection.subset_groups:
+            # If name matches name from the list subsets_to_apply, continue
+            if subsets.label in subsets_to_apply:
```

Comment:
so currently this will result in the last entry in `subsets_to_apply` being returned?  I think this logic would need to be completely rewritten to support multiple subsets anyways, so might just be worth requiring a string for now to avoid any unexpected results.